devices/Hx711/README.md

```diff
@@ -0,0 +1,82 @@
+# Hx711 (M5Stack WEIGHT)
+
+This is a library to interact with the Hx711 ADC. This is conveniently assembled and package in the M5Stack WEIGHT module.
+
+![weight_05.png](./weight_05.png)
+
+## Documentation
+
+* The module documentation can be found [here](https://docs.m5stack.com/en/unit/weight).
+* Datasheet for the Hx711 ADC can be found [here](https://m5stack.oss-cn-shenzhen.aliyuncs.com/resource/docs/datasheet/unit/HX711_en.pdf).
+
+## Usage
+
+**Important**: make sure you setup the SPI pins for ESP32 before creating the `Scale`. For this, make sure you install the `nanoFramework.Hardware.ESP32` NuGet:
+
+```csharp
+///////////////////////////////////////////////////////////////////////
+// When connecting to an ESP32 device, need to configure the SPI GPIOs
+// The following mapping is used in order to connect the WEIGTH module
+// to a M5Core device using the Grove port A
+// MOSI: connects to Grove pin 1
+// MISO: connects to Grove pin 2
+// CLOCK: connect to any free port as it's not used at all
+Configuration.SetPinFunction(21, DeviceFunction.SPI1_MOSI);
+Configuration.SetPinFunction(22, DeviceFunction.SPI1_MISO);
+Configuration.SetPinFunction(23, DeviceFunction.SPI1_CLOCK);
+```
+
+For other devices, like STM32, please make sure you're using the pins for the UART you want to use.
+
+```csharp
+// setup SPI connection settings
+// the clock value was adjusted in order to get the typical duration expected by the PD_SCK ~1us
+var spisettings = new SpiConnectionSettings(1, 19)
+{
+    ClockFrequency = 700_000
```

Comment:
reading it here but you may want to add a constant in the main class that contains this value. So it will make it easier for developers to use it.